workflow/controller/healthz.go

```diff
@@ -0,0 +1,57 @@
+package controller
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/argoproj/argo-workflows/v3/util/env"
+	"github.com/argoproj/argo-workflows/v3/workflow/common"
+)
+
+var (
+	age = env.LookupEnvDurationOr("HEALTHZ_AGE", 5*time.Minute)
+)
+
+// https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request
+// If we are in a state where there are any workflows that have not been reconciled in the last 2m, we've gone wrong.
+func (wfc *WorkflowController) Healthz(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	instanceID := wfc.Config.InstanceID
+	instanceIDSelector := func() string {
```

Comment:
@whynowy added instance ID, I had to make this a receiver func on `WorkflowController` because `main.go` does not know the instance ID